packages/@o3r/design/src/core/design-token/renderers/design-token-style.renderer.ts

```diff
@@ -22,9 +22,62 @@ export const computeFileToUpdatePath = (root = process.cwd(), defaultFile = 'sty
  * Compare Token variable by name
  * @param a first token variable
  * @param b second token variable
+ * @deprecated use {@link getTokenSorterByName} instead. Will be removed in v13.
  */
 export const compareVariableByName = (a: DesignTokenVariableStructure, b: DesignTokenVariableStructure): number => a.getKey().localeCompare(b.getKey());
 
+/**
+ * Compare Token variable by name
+ * @param _variableSet Complete set of the parsed Design Token
+ */
+export const getTokenSorterByName: TokenListTransform = (_variableSet) => (tokens) => tokens.sort((a, b) => a.getKey().localeCompare(b.getKey()));
+
+
+/**
+ * Compare Token variable by reference to write theme in reference order
+ * @param variableSet Complete set of the parsed Design Token
+ */
+export const getTokenSorterByRef: TokenListTransform = (variableSet) => {
+  /**
+   * Get the maximum number of reference levels for a Design Token, following its references
```

Comment:
Just to be sure I understand:
```
    "test": {
      "height": {
        "$extensions": {
          "o3rPrivate": true
        },
        "$value": 2.3,
        "$type": "number"
      },
      "width": {
        "$value": "{example.test.height}"
      },

      "border": {
        "$type": "border",
        "$value": {
          "color": "{example.color}",
          "style": "dashed",
          "width": "{example.test.width}"
        }
      }
```
border would return 2, width would return 1 and height 0?

Reply:
exactly